thriftbp/client_pool.go

```diff
@@ -429,6 +434,21 @@ func NewCustomClientPoolWithContext(
 	return newClientPool(ctx, cfg, genAddr, protoFactory, middlewares...)
 }
 
+const ThriftHostnameHeader = "thrift-hostname"
+
+func ThriftHostnameHeaderMiddleware(hostname string) thrift.ClientMiddleware {
```

Comment:
Please write doccomments for this function

especially in the doccomment we should mention that if they already set `ThriftHostname` to non-empty string in the config then this middleware will be automatically applied in their client pool and they do not need to explicitly set/add this middleware again.

we do allow users to add additional middlewares via `NewBaseplateClientPool`/`NewBaseplateClientPoolWithContext`/`NewCustomClientPool`, so we want to avoid the situation of people adding it twice.

Reply:
I'm also thinking of not exporting it. That would reduce confusion

Reply:
yea that works, too.